src/main/java/org/diatliuk/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,124 @@
+import org.diatliuk.bookstore.dto.order.item.OrderItemDto;
+import org.diatliuk.bookstore.enums.Status;
+import org.diatliuk.bookstore.exception.EntityNotFoundException;
+import org.diatliuk.bookstore.mapper.OrderItemMapper;
+import org.diatliuk.bookstore.mapper.OrderMapper;
+import org.diatliuk.bookstore.model.CartItem;
+import org.diatliuk.bookstore.model.Order;
+import org.diatliuk.bookstore.model.OrderItem;
+import org.diatliuk.bookstore.model.User;
+import org.diatliuk.bookstore.repository.book.BookRepository;
+import org.diatliuk.bookstore.repository.cart.CartItemRepository;
+import org.diatliuk.bookstore.repository.cart.ShoppingCartRepository;
+import org.diatliuk.bookstore.repository.order.OrderItemRepository;
+import org.diatliuk.bookstore.repository.order.OrderRepository;
+import org.diatliuk.bookstore.service.OrderService;
+import org.diatliuk.bookstore.service.ShoppingCartService;
+import org.diatliuk.bookstore.service.UserService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+    private final UserService userService;
+    private final ShoppingCartService cartService;
+    private final ShoppingCartRepository cartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final BookRepository bookRepository;
+
+    @Override
+    public OrderDto save(CreateOrderRequestDto requestDto) {
+        User authenticatedUser = userService.getAuthenticatedUser();
+        ShoppingCartDto shoppingCartDto = cartService.get();
+
+        Order order = createOrder(authenticatedUser, shoppingCartDto);
+        order.setShippingAddress(requestDto.getShippingAddress());
+        Order savedOrder = orderRepository.save(order);
+
+        Set<CartItem> cartItems = cartRepository
+                .getShoppingCartByUserId(authenticatedUser.getId())
+                .getCartItems();
+        List<OrderItem> orderItems = cartItems
+                .stream()
+                .map(orderItemMapper::cartItemToOrderItem)
+                .peek(orderItem -> orderItem.setOrder(savedOrder))
+                .toList();
+
+        savedOrder.setOrderItems(Set.copyOf(orderItemRepository.saveAll(orderItems)));
+        cartItemRepository.deleteAll(cartItems);
+        return orderMapper.toDto(savedOrder);
+    }
+
+    @Override
+    public List<OrderDto> getAll(Pageable pageable) {
+        return orderRepository.findAll(pageable)
+                .stream()
+                .map(orderMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public OrderDto update(Long id, UpdateOrderStatusRequestDto requestDto) {
```

Comment:
What if user will specify id of an order which do not belong to him? You should validate not only access to the particular endpoint, but also access of a user to the particular resource (make sure that in other methods and services you don't forget about it)

Reply:
Okay, I'll check my services, but usually I didn't add the validation because of some methods related to admin role